### ISSUE TITLE
Updating resources cached by the Service Worker requires an application restart and _two_ page reloads

### ISSUE THREAD

**Issue by vlukashov** [vaadin/flow#9310]
When updating an image, a web app manifest, or another resource cached by the Service Worker to be available offline, the update is not visible in the browser until the server is restarted (in development mode), and only **on the second** full page reload after the restart. This is unexpected.

### Minimal reproducible example
1. download a Vaadin 18 starter project from https://start.vaadin.com
1. override the version of Flow to be `5.0.offline-SNAPSHOT`
1. enable offline caching for `images/logo.png`:  `@PWA(name = "My Project", shortName = "My Project", offlineResources = {"images/logo.png"})`
1. start the app in development mode with `mvn`
1. replace the content of the `images/logo.png` file with ![logo.png](https://images.zenhubusercontent.com/5a572ab64b5806bc2bc2536e/f9780310-3d7f-46fb-bb98-f151350a8af6)
1. reload the browser page (see that the logo remains the same), reload multiple times
1. stop and restart the server (`mvn` again)
1. reload the browser page one (see that the logo remains the same)
1. reload the browser page the second time (see that now the new logo is visible)

### Expected behavior
When in development mode, I expect that updates to a cached image or the web app manifest are available immediately after the hot page reload.

When in production mode, I expect an 'update is available' UX pattern to be visible in the application to signal users that the assets loaded from the service worker cache got new versions, and the app needs a reload.

### Versions:
    - Vaadin / Flow version: 18.0.0.alpha1 / 5.0.offline-SNAPSHOT
    - Java version: 11
    - OS version: macOS 10.14.6
    - Browser version (if applicable): Brave 1.16.68 / Chromium 86.0.4240.183.

**Comment by haijian-vaadin** [comment]
Agreed that Gizmo should ask the service worker to update itself before reloading the page.

Test: multiple tabs

Production mode would be a separate task.
